### PR TITLE
Implement PUT /projects/:id/deploy_keys/:key_id

### LINKED ISSUE #543: Implement update deploy key method
Currently, there's no way to update an existing deploy key. According to API documentation there's an endpoint for that https://docs.gitlab.com/ee/api/deploy_keys.html#update-deploy-key

### PR DESCRIPTION
**Pull request by Quintasan**
This *should* work but I have no idea why rspec whines about my request stubs.

Closes #543

### CODE REVIEW

File: spec/gitlab/client/projects_spec.rb [hunk 1]
```diff
@@ -600,6 +600,47 @@
     end
   end
 
+  describe '.edit_deploy_key' do
+    context 'no options' do
+      before do
+        body = { title: 'New key name' }
+        stub_put('/projects/42/deploy_key/2', 'project_key').with(body: body)
```
Comment (NARKOZ) [inline spec/gitlab/client/projects_spec.rb]:
```suggestion
        stub_put('/projects/42/deploy_keys/2', 'project_key').with(body: body)
```

File: spec/gitlab/client/projects_spec.rb [hunk 2]
```diff
@@ -600,6 +600,47 @@
...
+      it 'returns the correct updated information' do
+        expect(@project_deploy_key).to be_a Gitlab::ObjectifiedHash
+        expect(@project_deploy_key.title).to eq 'New key name'
+      end
+    end
+
+    context 'some options' do
+      before do
+        body = { title: 'New key name', can_push: true }
+        stub_put('/projects/42/deploy_key/2', 'project_key').with(body: body)
```
Comment (NARKOZ) [inline spec/gitlab/client/projects_spec.rb]:
```suggestion
        stub_put('/projects/42/deploy_keys/2', 'project_key').with(body: body)
```

**Comment by Quintasan** [comment]
@NARKOZ Thanks, I must have gone blind or something :sweat_smile:

**Comment by NARKOZ** [comment]
Great. Thank you 👍

**Pull request merged**